packages/caip/src/adapters/banxa/index.ts

```diff
@@ -47,3 +50,18 @@ export const getSupportedBanxaAssets = () =>
     CAIP19,
     ticker
   }))
+
+const caip2ToBanxaBlockchainCodeMap: Record<CAIP2, string> = {
+  'eip155:1': 'ETH',
+  'bip122:000000000019d6689c085ae165831e93': 'BTC',
+  'cosmos:cosmoshub-4': 'COSMOS'
+}
+
+export const getBanxaBlockchainFromBanxaAssetTicker = (asset: string): string => {
+  const assetCAIP19 = banxaTickerToCAIP19(asset.toLowerCase())
+  if (!assetCAIP19)
+    throw new Error(`getBanxaBlockchainFromBanxaAssetTicker: ${asset} is not supported`)
+  const { chain, network } = fromCAIP19(assetCAIP19)
+  const caip2 = toCAIP2({ network, chain })
+  return caip2ToBanxaBlockchainCodeMap[caip2]
+}
```

Comment:
In general, I don't like `get` (or `is`) functions to `throw` when something isn't found. I'd rather just return `undefined`. I assume this is following a pattern from another file in this package. If that's the case, I get it.

Reply:
@cjthompson that `throw` part is suggested by @gomesalexandre in his review.
this function gets banxaAssetTicker which they're values of `CAIP19ToBanxaTickerMap`,
so the first step is to convert Banxa asset ticker to caip19,
then get chain and network from caip19,
then create caip2 with those variables, and find the related banxa blockchain_code from `caip2ToBanxaBlockchainCodeMap`.
only "btc", "eth" and "cosmos" work with your suggested implementation.

Reply:
> only "btc", "eth" and "cosmos" work with your suggested implementation.

those values are the only acceptable inputs in your implementation also. I don't see the difference.

Reply:
any Banxa ticker can be passed to this function, this is the list of the tickers: https://github.com/shapeshift/lib/blob/b0dd979413f400db08bb360992148d49a4728478/packages/caip/src/adapters/banxa/index.ts#L4-L33

Reply:
> any Banxa ticker can be passed to this function, this is the list of the tickers:

`as const` would probably help here. It narrows down the typing of this object to these specific keys/values primitives (strings) vs. any string as key, any string as value as we have currently. If we go that route, we'll need a type guard as well, since with the web logic, we will try all of our supported assetIds against `CAIP19ToBanxaTickerMap` which would produce type errors in case of an assetId that's not in this object.

Something like:
```typescript
const isValidBanxaAssetId = (assetId: string): assetId is keyof typeof CAIP19ToBanxaTickerMap => Boolean(CAIP19ToBanxaTickerMap[assetId as keyof typeof CAIP19ToBanxaTickerMap])
```
([TS Playground demo](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhhAwuaB5ARgKwKbFgXhgG8AoGcmAMxBAC4YBydOAJwYBoyLmAvehgI4BXAB4MSAX3gQYoSFBKK50GAEsIANTgAbVQBMAgsGDYIEECxiEAFHGOnzLetBaqwAcwCU9OybMW1GQBrbABPEEoYKFCAB2wI6WR5DBw8KwA+GAAhGm1sODBbJBQoFNwoAG1fBwCEGBDwyOi4hIQktCxygF1PRRJlWHUAMRotXT0rQLH9Iz9HawZqEAZe5RA8gDptEHdrImHRnX0JVZLAkaEWaIAVAHcQaYnCdUfZmpYF6kub+5X+lHW2C2Oz2B2+oTuDyOehOQA))

If we have such safer access, then we wouldn't need to throw anymore. Type-level safety is probably a better option, thanks @cjthompson!